src/test/java/io/jenkins/plugins/analysis/core/graphs/HealthSeriesBuilderTest.java

```diff
@@ -0,0 +1,425 @@
+package io.jenkins.plugins.analysis.core.graphs;
+
+import java.util.List;
+
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.util.Lists;
+import org.jfree.data.category.CategoryDataset;
+import org.junit.jupiter.api.Test;
+
+import edu.hm.hafner.analysis.Priority;
+import io.jenkins.plugins.analysis.core.quality.AnalysisBuild;
+import io.jenkins.plugins.analysis.core.quality.HealthDescriptor;
+import io.jenkins.plugins.analysis.core.quality.StaticAnalysisRun;
+import static org.mockito.Mockito.*;
+
+/**
+ * Tests the class {@link HealthSeriesBuilder}.
+ *
+ * @author Joscha Behrmann
+ */
+@SuppressWarnings("Duplicates")
+class HealthSeriesBuilderTest {
+
+    /**
+     * When given empty runs as input, no data should be generated when
+     * using build-number as domain.
+     */
+    @Test
+    void emptyAnalysisInputOnBuildNumberShouldReturnEmptyDataSet() {
+        GraphConfiguration cfg = createGraphConfig();
+        SeriesBuilder builder = new HealthSeriesBuilder(createDisabledHealthDescriptor());
+
+        SoftAssertions.assertSoftly(softly -> {
+            when(cfg.useBuildDateAsDomain()).thenReturn(false);
+            CategoryDataset dataSet = builder.createDataSet(cfg, Lists.newArrayList());
+            softly.assertThat(dataSet.getRowCount()).isEqualTo(0);
+            softly.assertThat(dataSet.getColumnCount()).isEqualTo(0);
+        });
+    }
+
+    /**
+     * When given empty runs as input, no data should be generated when
+     * using build-date as domain.
+     */
+    @Test
+    void emptyAnalysisInputOnDateDomainShouldReturnEmptyDataSet() {
+        GraphConfiguration cfg = createGraphConfig();
+        SeriesBuilder builder = new HealthSeriesBuilder(createDisabledHealthDescriptor());
+
+        SoftAssertions.assertSoftly(softly -> {
+            CategoryDataset dataSet2 = builder.createDataSet(cfg, Lists.newArrayList());
+            softly.assertThat(dataSet2.getColumnCount()).isEqualTo(0);
+            softly.assertThat(dataSet2.getRowCount()).isEqualTo(0);
+        });
+    }
+
+    /**
+     * Results that are too old shouldn't be added to the data set.
+     */
+    @Test
+    void oldResultsShouldNotBeAdded() {
+        GraphConfiguration cfg = createGraphConfig();
+        ResultTime resTime = mock(ResultTime.class);
+        SeriesBuilder builder = new HealthSeriesBuilder(createDisabledHealthDescriptor(), resTime);
+
+        // Only first 3 runs are new enough to be considered
+        when(resTime.areResultsTooOld(any(), any())).thenReturn(false, false, false, true);
+
+        SoftAssertions.assertSoftly(softly -> {
+            CategoryDataset dataSet = builder.createDataSet(cfg, Lists.newArrayList(createAnalysisRuns()));
+            softly.assertThat(dataSet.getRowCount()).isOne();
+            softly.assertThat(dataSet.getColumnCount())
+                    .as("Old runs shouldn't be added to the dataset")
+                    .isEqualTo(2);
```

Comment:
Wieso kommt hier 2 raus wenn oben 3 mal false steht?

Reply:
Das liegt daran, dass von den ersten drei Runs die noch 'neu' genug sind (false, false, false) zwei den gleichen Buildtag haben, also später gemerged werden. Das sieht man hier wenn überhaupt erst beim Blick auf die von createAnalysisRuns() erzeugten Elemente. Hab das in dem neuen Commit verbessert.